lib/kafka_ex/protocol/metadata.ex

```diff
@@ -2,6 +2,8 @@ defmodule KafkaEx.Protocol.Metadata do
   alias KafkaEx.Protocol
   import KafkaEx.Protocol.Common
 
+  @default_api_version 0
```

Comment:
After your previous comment wondering if we should create a new metadata_v1 module? I have not thought deeply about this just throwing it out as I imagine the more versions of things we support to more clunky the constructing request and parsing of response may become...

Reply:
I think it's a relevant question. That's indeed the kind of question that require a deep understanding of this project, which I'm not sure I have yet.

Especially since 0.10.0.0, the client is supposed to ask the broker which API versions are supported (see "[Retrieving Supported API versions](https://kafka.apache.org/protocol.html#api_versions)"). The *"broker returns its full list of supported ApiKeys and versions"*, meaning KafkaEx should use that information to decide which API message and version to use (and allow)

This means that for version >= 0.10.0.0, KafkaEx's behaviour should be quite different than the current hard-coded mapping of versions -> code.

Reply:
I'll leave the decision on how to proceed to @joshuawscott and @dantswain both of whom have been more involved in the project more recently than I have, however I think we should have different version modules and after looking up the supported api version either reject the api-version provided by the user or pick the newest one we have a module for and use the module... last time I looked I think there are a couple of versions of the requests that will require some amount of work to implement them, we need as much help as we can get :)

Reply:
I'm wondering if maybe this discussion should happen in another issue, so that we can move forward with this merge request.  It seems the scale is much bigger than just adding support for a new message in the protocol. I'd be happy to help, of course !

Would that be OK ?

Reply:
@dantswain @joshuawscott what do yall think?

Reply:
@joshuawscott agreed.